Tools/CMake/AMReX_Options.cmake

```
@@ -192,81 +188,45 @@ print_option( ENABLE_ASSERTIONS )
 
 
 #
-# Profiling options
+# Profiling options  =========================================================
 #
 option( ENABLE_BASE_PROFILE "Enable basic profiling" OFF )
 print_option( ENABLE_BASE_PROFILE )
 
-option( ENABLE_TINY_PROFILE "NOT ENABLE_BASE_DEBUG" OFF)
+cmake_dependent_option( ENABLE_TINY_PROFILE "Enable tiny profiling" OFF
+   "NOT ENABLE_BASE_PROFILE" OFF)
 print_option( ENABLE_TINY_PROFILE )
 
-option( ENABLE_TRACE_PROFILE "Enable trace-profiling" OFF )
+cmake_dependent_option( ENABLE_TRACE_PROFILE "Enable trace-profiling" OFF
+   "ENABLE_BASE_PROFILE" OFF)
 print_option( ENABLE_TRACE_PROFILE )
 
 option( ENABLE_MEM_PROFILE   "Enable memory profiling" OFF )
 print_option( ENABLE_MEM_PROFILE )
 
-option( ENABLE_COMM_PROFILE  "Enable communicator-profiling" OFF )
+cmake_dependent_option( ENABLE_COMM_PROFILE  "Enable communicator-profiling" OFF
+   "ENABLE_BASE_PROFILE" OFF)
 print_option( ENABLE_COMM_PROFILE )
 
 option( ENABLE_BACKTRACE "Enable backtracing" OFF)
 print_option( ENABLE_BACKTRACE )
 
-option( ENABLE_PROFPARSER "Enable profile parser" OFF)
+cmake_dependent_option(ENABLE_PROFPARSER "Enable profile parser" OFF
+   "ENABLE_BASE_PROFILE;ENABLE_TRACE_PROFILE;ENABLE_AMRDATA" OFF)
 print_option( ENABLE_PROFPARSER )
 
-set ( TP_PROFILE "None" CACHE STRING "Third-party profiling options:<CRAYPAT,FORGE,VTUNE>")
+set( TP_PROFILE "None" CACHE STRING "Third-party profiling options:<CRAYPAT,FORGE,VTUNE>")
```

Straight from Craig Scott https://stackoverflow.com/a/39671055/2719194 we can set a GUI helper and validate against the values below with a list:
```suggestion
set(TP_PROFILE_VALUES None CRAYPAT FORGE VTUNE)
set(TP_PROFILE None CACHE STRING "Third-party profiling options")
set_property(CACHE TP_PROFILE PROPERTY STRINGS ${TP_PROFILE_VALUES})
```

---

Looks good to me! I would keep the available options, i.e. "Third-party profiling options:<CRAYPAT,FORGE,VTUNE>", in the doc string for cache variable TP_PROFILE.

---

Oh right, I though `ccmake` shows them but maybe that's just a GUI thingy or no feature at all.